managed/resource_cluster_db_audit_export_config.go

```diff
@@ -0,0 +1,394 @@
+		return
+	}
+
+	dbAuditExporterConfigSpec, err := getDbAuditExporterConfigSpec(plan)
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to obtain DbAuditExporterConfigSpec", GetApiErrorDetails(err))
+		return
+	}
+
+	response, _, err := apiClient.ClusterApi.AssociateDbAuditExporterConfig(ctx, accountId, projectId, clusterId).DbAuditExporterConfigSpec(*dbAuditExporterConfigSpec).Execute()
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to associate db audit log configuration to cluster", GetApiErrorDetails(err))
+		return
+	}
+
+	err = associateDbAuditExporterConfigWithRetry(ctx, apiClient, accountId, projectId, clusterId)
+
+	if err != nil {
+		errorSummary := fmt.Sprintf("Unable to associate db audit log config to cluster: %s", clusterId)
+		resp.Diagnostics.AddError(errorSummary, "The operation timed out waiting for db audit log config cluster association.")
+		return
+	}
+
+	configId := response.Data.Info.Id
+	plan.ConfigID.Value = configId
+
+	dae, readOK, err := resourceAssociateDbAuditLogConfigClusterRead(ctx, accountId, projectId, clusterId, configId, apiClient)
+	if !readOK {
+		resp.Diagnostics.AddError("Unable to read the state of Db Audit log config cluster association ", GetApiErrorDetails(err))
+		return
+	}
+
+	diags := resp.State.Set(ctx, &dae)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func associateDbAuditExporterConfigWithRetry(ctx context.Context, apiClient *openapiclient.APIClient, accountId, projectId, clusterId string) error {
+	retryPolicy := retry.NewConstant(10 * time.Second)
+	retryPolicy = retry.WithMaxDuration(2400*time.Second, retryPolicy)
+	return retry.Do(ctx, retryPolicy, func(ctx context.Context) error {
+		asState, readInfoOK, message := getTaskState(accountId, projectId, clusterId, openapiclient.ENTITYTYPEENUM_CLUSTER, openapiclient.TASKTYPEENUM_MODIFY_DB_AUDIT_EXPORT_CONFIG, apiClient, ctx)
+		if readInfoOK {
+			if asState == string(openapiclient.TASKACTIONSTATEENUM_SUCCEEDED) {
+				return nil
+			}
+			if asState == string(openapiclient.TASKACTIONSTATEENUM_FAILED) {
+				return fmt.Errorf("unable to associate cluster with db audit log configuration, operation failed")
+			}
+		} else {
+			return retry.RetryableError(errors.New("Unable to get check db audit log configuration cluster association: " + message))
+		}
+		return retry.RetryableError(errors.New("db audit log config is being associated to the cluster"))
+	})
+}
+
+func resourceAssociateDbAuditLogConfigClusterRead(ctx context.Context, accountId string, projectId string, clusterId string, configId string, apiClient *openapiclient.APIClient) (dbAuditExporterConfig DbAuditExporterConfig, readOK bool, err error) {
+	listDbAuditExporterConfigResp, _, err := apiClient.ClusterApi.ListDbAuditExporterConfig(ctx, accountId, projectId, clusterId).Execute()
+	if err != nil {
+		return dbAuditExporterConfig, false, err
```

Comment:
Shouldn't be a GetErrorApi ? instead of err ?

Reply:
Fixed